### ISSUE TITLE
Validate absolute DNS name requirement on StaticDNS CNAME

### ISSUE THREAD

**Issue by jhg03a** [apache/trafficcontrol#4743]
## I'm submitting a ...

- [ ] bug report
- [ ] new feature / enhancement request
- [x] improvement request (usability, performance, tech debt, etc.)
- [ ] other 

## Traffic Control components affected ...

- [ ] CDN in a Box
- [ ] Documentation
- [ ] Grove
- [ ] Traffic Control Client
- [ ] Traffic Monitor
- [x] Traffic Ops
- [ ] Traffic Ops ORT
- [x] Traffic Portal
- [ ] Traffic Router
- [ ] Traffic Stats
- [ ] Traffic Vault
- [ ] unknown

## Current behavior:

When I was creating a Static DNS CNAME record, I added `domain.foo.com` to a new DS and snapshotted.
This causes the snapshot to fail to apply with no user feedback except to go dig around in TR logs.

## Expected / new behavior:

Please remove the footgun of forgetting the trailing dot in the CNAME address and better safeties to prevent it from becoming part of a snapshot.

## Minimal reproduction of the problem with instructions:

1. Add a staticDNS CNAME record with address lacking the final period.
2. Snapshot
3. Observe the Snapshot fails to apply
4. Add a trailing period to the address field of record from 1.
5. Snapshot
6. Observe things work after head banging

## Anything else:

```
WARN  2020-05-30T01:57:54.921 [New I/O worker #8] com.comcast.cdn.traffic_control.traffic_router.core.util.PeriodicResourceUpdater - 'ccr.testds.cdn.domain.com' is not an absolute name
org.xbill.DNS.RelativeNameException: 'domain.foo.com' is not an absolute name
        at org.xbill.DNS.Record.checkName(Record.java:721)
        at org.xbill.DNS.SingleNameBase.<init>(SingleNameBase.java:33)
        at org.xbill.DNS.SingleCompressedNameBase.<init>(SingleCompressedNameBase.java:23)
        at org.xbill.DNS.CNAMERecord.<init>(CNAMERecord.java:28)
        at com.comcast.cdn.traffic_control.traffic_router.core.dns.ZoneManager.addStaticDnsEntries(ZoneManager.java:703)
```

**Comment by ocket8888** [comment]
So is the preferred solution to make it so that a CNAME _can_ be added without a trailing <kbd>.</kbd> or so that such CNAMEs would be rejected by the API?

**Comment by jhg03a** [comment]
Yes.  Implementer's choice.

**Comment by jhg03a** [comment]
Or at least update the documentation with a CNAME example showing it's a requirement

**Comment by jhg03a** [comment]
Reading stack traces and digging through 3rd party library source code to find out what "Absolute" means is not something operators can afford to be doing during an outage.

**Comment by mitchell852** [comment]
when you create a federation cname, you have to add the trailing dot. i say we keep it consistent and just require the trailing dot.

**Comment by mitchell852** [comment]
also, when fixing this form. the "address" field is ambiguous and apparently has some rules that must be followed or you'll get this error in TR:

`org.xbill.DNS.RelativeNameException: 'foo' is not an absolute name`

**Comment by mitchell852** [comment]
also, i added the `bug` label because missing validation feels like a bug to me.

**Comment by mitchell852** [comment]
I also changed the title @jhg03a - hope you don't mind.

**Comment by jhg03a** [comment]
> Yes.  Implementer's choice.

That's fine. It just needs a helpful message to go with and not just "denied"

**Comment by mitchell852** [comment]
I put the `major` label on this as it will block a snapshot.

**Comment by rimashah25** [comment]
jh03a: Does the trailing period affect only DNS type cname_record or the other 3 as well?

**Comment by jhg03a** [comment]
CNAME is where I found it. In theory it would be anywhere we specify the Dns name of a record that uses that library to validate.

**Comment by mitchell852** [comment]
fixed per #4934
